Closes #2326 check_type documentation is updated for argument details.

Lina2689 opened the pull request:
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Review the [Cheat Sheet](https://github.com/pharmaverse/admiral/blob/main/inst/cheatsheet/admiral_cheatsheet.pdf). Make any required updates to it by editing the file `inst/cheatsheet/admiral_cheatsheet.pptx` and re-upload a PDF version of it to the same folder.
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

bms63 reviewed (commented):
Thanks @Lina2689 - I'm just wondering if we need a test set up to make sure this is behaving appropriately??  @bundfussr WDYT??

Perhaps there is already one??

bundfussr commented on R/derive_merged.R:
```diff
@@ -127,6 +127,10 @@
 #'   if the observations of the (restricted) additional dataset are not unique
 #'   with respect to the by variables and the order.
 #'
+#'   If the `check_type` argument is specified, `order` must be set to a non-null value.
```
This is not true. I would remove this sentence.

bundfussr commented on R/derive_merged.R:
```diff
@@ -127,6 +127,10 @@
 #'
+#'   If the `check_type` argument is specified, `order` must be set to a non-null value.
+#'
+#'   If the `order` argument is not specified, the `check_type` argument is ignored.
```
We should say what "ignored" means. I would add something like "I.e., if the observations of the (restricted) additional dataset are not unique with respect to the by variables, an error is issued."

Lina2689 replied:
Thanks @bundfussr , I will modify the sentence.

bundfussr commented:
> Thanks @Lina2689 - I'm just wondering if we need a test set up to make sure this is behaving appropriately?? @bundfussr WDYT??
> 
> Perhaps there is already one??

Yes, a test for the case that `order` is not specified and duplicates in `dataset_add` with respect to `by_vars` is a good idea.

Lina2689 commented:
@bundfussr & @bms63 , I have added test cases when order is not specified and also updated the description in header. Can you please review and let me know if anything needs to be updated.

bundfussr commented on tests/testthat/test-derive_merged.R:
```diff
@@ -631,3 +631,48 @@ test_that("derive_var_merged_summary Test 23: deprecation warning", {
+      order = NULL,
+      check_type = "error"
+    ),
+    regexp = NULL
```
Please specify a value for `regexp` which ensures that the expected error is issued. Otherwise, any error is accepted.

bms63 reviewed (approved):
Thanks @Lina2689 !!!!!!  

This is your first official PR?!?!  Congrats if so!!!!!!!! A big accomplishment!!
